### ISSUE TITLE
[Rolling Stock Stars] Corporations in receivership close wrong companies

### ISSUE THREAD

**Issue by beorn7** [tobymao/18xx#8585]
**What's your Game ID?**

https://18xx.games/game/102758?action=581

**How to Reproduce the Bug?**

After the action linked above, the VM corporation has closed companies OL and KME, retaining BME.

However, the rules mandate that VM has to retain OL in that case.

Also, the wording in the logs is misleading. It reads:
```
[11:51] OL (VM - Receivership) has negative income
[11:51] OL (VM) closes
```

However, corporations in receivership do not close due to negative income but following a schedule (close reds when CoO is $4, close orange when CoO is $7). This _might_ be implemented correctly. I'm just saying that the log message is misleading.

**Which Browser / Platform?**
- Browser Version: Google Chrome	108.0.5359.94 (Official Build) (64-bit)
- Platform: Desktop

**Comment by roseundy** [comment]
I think the 2nd issue is just a logging issue, since I remember implementing the schedule. I still need to look into the first issue. It will likely not be until this weekend (after final exams this week).

**Comment by beorn7** [comment]
Good look with the exams.

It looks the 1st issue is just inverted sorting (rather than starting with the most expensive company, start with the cheapest when closing).

I sincerely hope I'll soon get into 18xx.games development and can just submit PRs rather than issues for those things.

**Comment by roseundy** [comment]
Yes, issue #1 was due to sorting companies in descending order rather than ascending.

Fixed issue #2. Now prints:
KME (VM - Receivership) exceeds maximum cost of ownership
KME (VM) closes
